dropwizard-jetty/src/main/java/io/dropwizard/jetty/BiDiGzipHandler.java

```diff
@@ -87,18 +89,33 @@ private Inflater buildInflater() {
 
     private WrappedServletRequest wrapDeflatedRequest(HttpServletRequest request) throws IOException {
         final Inflater inflater = buildInflater();
-        final InflaterInputStream input = new InflaterInputStream(request.getInputStream(), inflater, inputBufferSize) {
-            @Override
-            public void close() throws IOException {
-                super.close();
-                localInflater.set(inflater);
-            }
-        };
-        return new WrappedServletRequest(request, input);
+        try {
+            final InflaterInputStream input = new InflaterInputStream(request.getInputStream(), inflater, inputBufferSize) {
+                @Override
+                public void close() throws IOException {
+                    super.close();
+                    localInflater.set(inflater);
+                }
+            };
+            InputStream exceptionHandlingInput = new ZipExceptionHandlingInputStream(input, DEFLATE);
+            return new WrappedServletRequest(request, exceptionHandlingInput);
+        } catch (ZipException e) {
+            throw ZipExceptionHandlingInputStream.buildBadDataException(DEFLATE, e);
+        } catch (EOFException e) {
+            throw ZipExceptionHandlingInputStream.buildPrematureEofException(DEFLATE, e);
+        }
     }
 
     private WrappedServletRequest wrapGzippedRequest(HttpServletRequest request) throws IOException {
-        return new WrappedServletRequest(request, new GZIPInputStream(request.getInputStream(), inputBufferSize));
+        try {
+            GZIPInputStream input = new GZIPInputStream(request.getInputStream(), inputBufferSize);
```

Comment:
Nit: please declare `input` as `final` for consistency with the rest of the codebase.